src/Serializers/QueryResultSerializer.php

```diff
@@ -56,6 +56,39 @@ public function isSerializerFor( $queryResult ) {
 	}
 
 	/**
+	 * Get display title of a page.
+	 *
+	 * @param int $pageId page ID of page being queried
+	 *
+	 * @return string|bool display title value or false if not found
+	 */
+	public static function getDisplayTitle( $pageId ) {
+
+		if ( ! is_numeric( $pageId ) || $pageId < 1 ) {
+
+			return false;
+
+		}
+
+		$dbr = wfGetDB( DB_SLAVE );
```

Comment:
I'm not so happy about the DB access in the QueryResultSerializer (which is untested) and should probably be handled by a service lookup class. In general I would see the Title object to actually have to information available since you rely on a convention introduced in [0] to provide/store the information.

Also, `QueryResultSerializer` is one of the performance drainer doing an uncached DB read doesn't make this class faster.

[0] https://gerrit.wikimedia.org/r/#/c/246246/